src/map/pc.c

```diff
@@ -5320,6 +5320,9 @@ static int pc_useitem(struct map_session_data *sd, int n)
 	// Update item use time.
 	sd->canuseitem_tick = tick + battle_config.item_use_interval;
 
+	if (nameid == ITEMID_MEGAPHONE)
+		sd->state.using_megaphone = 1;
```

Comment:
hm, using flag is set if fixed id was used, but unset flag only if script command used.
probably better call some script command in item use script for enable megaphone flag before ``input @megaphone$;``?

This will remove hard coded item id.

Reply:
I dislike the hard coded item ID, too. I'll try finding a better solution.

Reply:
Unfortunately I need the item ID prior to the execution of the item script.
I'll extend the item DB code in the next release to get rid of most of all those hard coded item IDs but for now we have to live with it...